main.go

```diff
@@ -156,9 +162,15 @@ Available arguments:
 			log.Println("Language Description: ", langDesc)
 		}
 
+		if stack != nil {
```

Comment:
How can this ever fail to be `nil`?  The `stack :=` is done before flag-parsing and defines the place to store the `-stack`, and will be non-nil even if the stack is not supplied.

I think you want to be creating `stack` the same way as all the other vars, and using `!= ""` here instead, to check for having been supplied.